acceptancetests/assess_model_migration_versions.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+"""Test migrating models between controllers of increasing versions."""
+
+from __future__ import print_function
+
+import argparse
+from distutils.version import (
+    LooseVersion,
+    StrictVersion
+    )
+import logging
+import os
+import sys
+
+from assess_model_migration import (
+    _new_log_dir,
+    assert_model_migrated_successfully,
+    deploy_simple_server_to_new_model,
+    migrate_model_to_controller,
+)
+from jujupy.client import get_stripped_version_number
+from deploy_stack import (
+    BootstrapManager,
+    client_from_config,
+    get_random_string,
+    )
+from utility import (
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+
+log = logging.getLogger('assess_model_migration_versions')
+
+
+def assess_model_migration_versions(stable_bsm, devel_bsm, args):
+    with stable_bsm.booted_context(args.upload_tools):
+        stable_bsm.client.enable_feature('migration')
```

Comment:
This would be a useful check in assert_stable_juju_suitable_for_testing, since in theory the older stable version might be missing the ability to migrate and thus should fail in that assertion.